packages/sanity/src/_internal/cli/actions/app/buildAction.ts

```diff
@@ -0,0 +1,226 @@
+  context: CliCommandContext,
+  overrides?: {basePath?: string},
+): Promise<{didCompile: boolean}> {
+  const timer = getTimer()
+  const {output, prompt, workDir, cliConfig, telemetry = noopLogger} = context
+  const flags: BuildSanityAppCommandFlags = {
+    'minify': true,
+    'stats': false,
+    'source-maps': false,
+    ...args.extOptions,
+  }
+
+  /**
+   * Unattended mode means that if there are any prompts it will use `YES` for them but will no change anything that doesn't have a prompt
+   */
+  const unattendedMode = Boolean(flags.yes || flags.y)
+  const defaultOutputDir = path.resolve(path.join(workDir, 'dist'))
+  const outputDir = path.resolve(args.argsWithoutOptions[0] || defaultOutputDir)
+
+  const autoUpdatesEnabled = shouldAutoUpdate({flags, cliConfig})
+
+  const installedSdkVersion = await readModuleVersion(context.workDir, '@sanity/sdk-react')
+
+  if (!installedSdkVersion) {
+    throw new Error(`Failed to find installed @sanity/sdk-react version`)
+  }
+  // Get the version without any tags if any
+  const coercedSanityVersion = semver.coerce(installedSdkVersion)?.version
+  if (autoUpdatesEnabled && !coercedSanityVersion) {
+    throw new Error(`Failed to parse installed Sanity version: ${installedSdkVersion}`)
+  }
+  const version = encodeURIComponent(`^${coercedSanityVersion}`)
+  const autoUpdatesImports = getStudioAutoUpdateImportMap(version)
+
+  if (autoUpdatesEnabled) {
+    output.print(`${info} Building with auto-updates enabled`)
+
+    // Check the versions
+    const result = await compareStudioDependencyVersions(autoUpdatesImports, workDir)
+
+    // If it is in unattended mode, we don't want to prompt
+    if (result?.length && !unattendedMode) {
+      const shouldContinue = await prompt.single({
+        type: 'confirm',
+        message: chalk.yellow(
+          `The following local package versions are different from the versions currently served at runtime.\n` +
+            `When using auto updates, we recommend that you test locally with the same versions before deploying. \n\n` +
+            `${result.map((mod) => ` - ${mod.pkg} (local version: ${mod.installed}, runtime version: ${mod.remote})`).join('\n')} \n\n` +
+            `Continue anyway?`,
+        ),
+        default: false,
+      })
+
+      if (!shouldContinue) {
+        return process.exit(0)
+      }
+    }
+  }
+
+  const envVarKeys = getSanityEnvVars()
+  if (envVarKeys.length > 0) {
+    output.print(
+      '\nIncluding the following environment variables as part of the JavaScript bundle:',
+    )
+    envVarKeys.forEach((key) => output.print(`- ${key}`))
+    output.print('')
+  }
+
+  let shouldClean = true
+  if (outputDir !== defaultOutputDir && !unattendedMode) {
+    shouldClean = await prompt.single({
+      type: 'confirm',
+      message: `Do you want to delete the existing directory (${outputDir}) first?`,
+      default: true,
+    })
+  }
+
+  // Determine base path for built studio
+  let basePath = '/'
```

Comment:
Do we need basepath logic in apps?

Reply:
hmm, it doesn't hurt to have it? Might be annoying to remove now and add it later if someone asks?